programs/aes/aescrypt2.c

```diff
@@ -29,9 +29,12 @@
 #include "mbedtls/platform.h"
 #else
 #include <stdio.h>
-#define mbedtls_fprintf    fprintf
-#define mbedtls_printf     printf
-#endif
+#include <stdlib.h>
+#define mbedtls_fprintf         fprintf
+#define mbedtls_printf          printf
+#define MBEDTLS_EXTI_SUCCESS    EXIT_SUCCESS
```

Comment:
`MBEDTLS_EXTI_SUCCESS`? Why not `MBEDTLS_EXIT_SUCCESS`? This needs fixing in many places. How does this build? What testing has been done?

Reply:
>MBEDTLS_EXTI_SUCCESS? Why not MBEDTLS_EXIT_SUCCESS? This needs fixing in many places. 

Sorry about that. You are right, there is a typo here and the mistake is repeated in all the other files because I largely copy-pasted the same definitions across the examples.

> How does this build?

By default, MBEDTLS_PLATFORM_C is defined, so this code is almost never used.

> What testing has been done?

I tested each of the applications with the default config.h that has MBEDTLS_PLATFORM_C so this code was never used. When MBEDTLS_PLATFORM_C is defined, the applications compiled without problems. I mainly tested the application's functionality, but this was in no way thorough as there are 40+ examples. Also, I am not aware of any automated tests to check the examples...

Now that I think about it, I am not even sure why the examples need these defines. I think platform.h defines these macros to something regardless of whether MBEDTLS_PLATFORM_C is or isnt defined and the default values are pretty sensible e.g:

[mbedtls_printf -> printf](https://github.com/ARMmbed/mbedtls/blob/development/include/mbedtls/platform.h#L191)
MBEDTLS_EXIT_SUCCESS -> 0
MBEDTLS_EXIT_FAILURE -> 1

Also, I think these default values for the macros need to be correct, otherwise tests like [this](https://github.com/ARMmbed/mbedtls/blob/development/tests/scripts/all.sh#L593) would probably fail as the library also has dependencies on these standard library symbols. @Patater, I will try replacing this code for `#include "mbedtls/platform.h"`...